src/module/actor/character/document.ts

```diff
@@ -1169,44 +1170,66 @@ class CharacterPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
             }
         }
 
-        // Get best weapon potency
-        const weaponPotency = (() => {
-            const potency = attackDomains
+        // Handle item or potency bonus to attack rolls
+        {
+            const grade = weapon.system.grade;
+            const potencyRune = weapon.system.runes.potency;
+
+            // Get all weapon potency synthetics. These don't work for sf2e unless the type is potency (ABP)
+            const potencySynthetics = attackDomains
                 .flatMap((key) => fu.deepClone(synthetics.weaponPotency[key] ?? []))
-                .filter((wp) => wp.predicate.test(initialRollOptions));
-
-            if (weapon.system.runes.potency > 0) {
-                potency.push({
-                    label: "PF2E.Item.Weapon.Rune.Potency",
-                    bonus: weapon.system.runes.potency,
-                    type: "item",
-                    predicate: new Predicate(),
+                .filter((wp) => wp.predicate.test(initialRollOptions) && (!grade || wp.type === "potency"));
+            const bestSynthetic = potencySynthetics.length
+                ? potencySynthetics.reduce((highest, current) => (highest.bonus > current.bonus ? highest : current))
+                : null;
+
+            // Handle weapon attack from the tracking trait (which isn't specific to tech/analog)
+            // If found, filter out inferior tracking traits from the weapon in case there are duplicates
+            const trackingTraits = ["tracking-1", "tracking-2", "tracking-3"] as const;
+            const tracking = trackingTraits.findLastIndex((t) => weapon.system.traits.value.includes(t)) + 1;
+            weapon.flags.pf2e.tracking = tracking;
+            if (tracking > 0) {
+                weapon.system.traits.value = weapon.system.traits.value.filter(
+                    (t) => t === `tracking-${tracking}` || !t.startsWith("tracking-"),
+                );
+            }
```

Comment:
Not crazy about using a flag for a core trait, and storing the index doesn't seem very useful, either.

Reply:
The flag is for rule elements to use. If we're willing to say tracking = item bonus, then we can drop it!

I'll remove it since I wasn't a fan of that and would prefer to ignore the finer details. If a soldier wants to drink a pf2e mutagen, yeah sure go for it.

Reply:
For PR posterity we plan on putting it in `weapon.system.traits.tracking` or `weapon.system.traits.data.tracking` but I think that's better suited for a followup PR so I removed it from here.